### DISCUSSION TITLE
Calling a service within a Python script

### DISCUSSION THREAD

**Discussion by lpmrz** [ctu-mrs/mrs_uav_system#107] [Q&A] [answered]
Hello,

Let's say I want to call a service within a Python script, namely /uav1/control_manager/start_trajectory_tracking. How can I achieve this?

Thank you very much.

edit: I tried the solution below but failed. I didn't understand why it needs an argument since when I check if it needs an argument it returns false.

![My code](https://cdn.discordapp.com/attachments/867626985144320026/878584065400373248/Screenshot_from_2021-08-21_13-18-11.png)

![The error](https://cdn.discordapp.com/attachments/867626985144320026/878584078943797268/Screenshot_from_2021-08-21_13-18-35.png)

edit#2 : I solved my problem.

**Comment by klaxalk** [comment, marked as answer]
Hello, this is a standard ROS and Python question. Please, google first, then search for examples, and then ask questions. Calling service requires setting a service client first.

**Reply by lpmrz** [reply]
Hello, yes I was too hasty to open an issue, I solved it by googling. Thank you for your reply.
